app/services/uphold/models/card.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'addressable/template'
+
+module Uphold
+  module Models
+    class Card < Client
+      include Initializable
+
+      # For more information about how these URI templates are structured read the explaination in the RFC
+      # https://www.rfc-editor.org/rfc/rfc6570.txt
+      PATH = "/v0/me/cards"
+
+      attr_accessor :address, :available, :balance, :currency, :id, :label, :lastTransactionAt, :normalized, :settings
+
+      def initialize(params = {})
+        super
+      end
+
+      # Finds a card given an id
+      #
+      # @param [UpholdConnection] connection The uphold connection to find.
+      # @param [string] iid The id of the card you want to find. By default searches for the address on UpholdConnection
+      #
+      # @return [Uphold::Models::Card] the card
+      def find(uphold_connection:, id: nil)
+        Rails.logger.info("Connection #{uphold_connection.id} is missing uphold_access_parameters") and return if uphold_connection.uphold_access_parameters.blank?
+        id = uphold_connection.address if id.blank?
+
+        endpoint = PATH + "/" + id
+        response = get(endpoint, {}, authorization(uphold_connection))
+
+        Uphold::Models::Card.new(JSON.parse(response.body))
+      end
+
+      # Searches for a card given the uphold_connection and a currency
+      #
+      # @param [UpholdConnection] connection The uphold connection to find.
+      # @param [string] currency The currency you want to find. By default searches for default_currency on UpholdConnection
+      #
+      # @return [Uphold::Models::Card[]] an array of found cards
+      def where(uphold_connection:, currency: nil)
+        Rails.logger.info("Connection #{uphold_connection.id} is missing uphold_access_parameters") and return if uphold_connection.uphold_access_parameters.blank?
+
+        query = "currency:" + (currency || uphold_connection.default_currency)
+
+        response = get(PATH, { q: query }, authorization(uphold_connection))
+
+        cards = []
+        if response.headers['Content-Encoding'].eql?('gzip')
+          sio = StringIO.new(response.body)
+          gz = Zlib::GzipReader.new(sio)
+          cards = JSON.parse(gz.read)
+        else
+          cards = JSON.parse(response.body)
+        end
+
+        cards = cards.map { |c| Uphold::Models::Card.new(c) } if cards.present?
+
+        cards
+      end
+
+      # Finds a card given the uphold_connection and a currency
```

Comment:
these docstrings are out of date

Reply:
https://github.com/brave-intl/publishers/pull/2062/commits/0b073f817a0e365a584bed4247f1a52848921e68